DEVELOPER.md

```diff
@@ -0,0 +1,114 @@
+# Notes for Development
+
+## Development Environment
+
+## Building from Source
+
+To build and test the driver, the standard cargo commands can be used from the root directory.
+
+For an unoptimized build with debugging information (most common):
+```
+cargo build
+```
+
+For an optimized build with debugging info (mac)
+```
+cargo build --features odbc-sys/iodbc,cstr/utf32 --profile=release-with-debug
+```
+And for an optimized build with debugging info (windows, linux)
+```
+cargo build --profile=release-with-debug
+```
+The resulting build files will be output to the `target` directory in the `debug` or `release` folders respectively.
+
+## Setting up the driver manager on MacOS
```

Comment:
This mirrors the windows version in the main `README.md` pretty closely, but given the relative state of both I thought this made sense in the developer readme for now. Happy to move it though if you think it belongs there instead!